src/cutadapt/__main__.py

```diff
@@ -1021,6 +1021,10 @@ def warn_if_en_dashes(args):
 
 
 def estimate_compression_threads(cores: int) -> Optional[int]:
+    if cores == 1:
+        # This sets xopen in a mode were only the main thread is used.
+        # No external programs will be used.
+        return 0
```

Comment:
With this, this function never returns 1, which I find a bit unexpected.

How about just `return max(0, min(cores - 1, 4))`?

Reply:
Done!